anaDACScan.py

```diff
@@ -0,0 +1,236 @@
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calDacCalFile = "{0}/{1}/calFile_calDac_{1}.txt".format(dataPath,chamber_config[oh])
+            calDacCalFileExists = os.path.isfile(calDacCalFile)
+            if not calDacCalFileExists:
+                print("Skipping OH{0}, detector {1}, missing CFG_CAL_DAC Calibration file:\n\t{2}".format(
+                    oh,
+                    chamber_config[oh],
+                    calDacCalFile))
+                ohArray = np.delete(ohArray,(oh))
+
+
+    if len(ohArray) == 0:
+        print('No OHs with a calFile, exiting.')
+        exit(1)
+           
+    # Initialize nested dictionaries
+    dict_dacScanResults = nesteddict()
+    dict_dacScanFuncs = nesteddict()
+
+    # Initialize a TGraphErrors for each vfat
+    for oh in ohArray:
+        for vfat in range(0,24):
+             dict_dacScanResults[oh][vfat] = r.TGraphErrors()
+             #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+             dict_dacScanResults[oh][vfat].GetXaxis().SetTitle(nameY)
+             dict_dacScanResults[oh][vfat].GetYaxis().SetTitle(nameX)
+
+    outputFiles = {}         
+             
+    for oh in ohArray:         
+        outputFiles[oh] = r.TFile(dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate+"/"+args.outfilename,'recreate')
+             
+    # Loop over events in the tree and fill plots
+    for event in dacScanFile.dacScanTree:
+        oh = event.link
+        vfat = event.vfatN
+
+        #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+        calibrated_DAC_value=calInfo[oh]['slope'][vfat]*event.dacValX+calInfo[oh]['intercept'][vfat]
+        calibrated_DAC_error=calInfo[oh]['slope'][vfat]*event.dacValX_Err+calInfo[oh]['intercept'][vfat]
+
+        if args.assignXErrors:
+            dict_dacScanResults[oh][vfat].SetPoint(dict_dacScanResults[oh][vfat].GetN(),event.dacValY,calibrated_DAC_value)
+            dict_dacScanResults[oh][vfat].SetPointError(dict_dacScanResults[oh][vfat].GetN()-1,event.dacValY_Err,calibrated_DAC_error)
+        else:    
+            dict_dacScanResults[oh][vfat].SetPoint(dict_dacScanResults[oh][vfat].GetN(),event.dacValY,calibrated_DAC_value)
+            dict_dacScanResults[oh][vfat].SetPointError(dict_dacScanResults[oh][vfat].GetN()-1,event.dacValY_Err,0)
+
+    # Fit the TGraphErrors objects    
+    for oh in ohArray:
+        for vfat in range(0,24):
+            dict_dacScanFuncs[oh][vfat] = r.TF1("DAC Scan Function","[0]*x+[1]")
+            dict_dacScanResults[oh][vfat].Fit(dict_dacScanFuncs[oh][vfat]) 
+
+    nominal = -1        
+
+    #from Tables 9 and 10 of the VFAT3 manual 
+    if nameX == "CFG_THR_ARM_DAC":
+        nominal = 3.2
+    else:
+        print("Error: unexpected value of nameX: '%s'"%nameX)
+        exit(1)
```

Comment:
Lines 189 to 196 probably just make this into a dictionary and store it in `utils/anaInfo.py`; the key value should be the register name.  Then you can just do a check if `nameX in dict.keys()` and then get the nominal value by passing `nameX` as the key.

Reply:
done